### PR TITLE
Add support for handing service url template

### PR DESCRIPTION
**Pull request by kyujin-cho**
This PR adds new `extrainfo` endpoint to stream API.
Check lablup/backend.ai-agent#152 and lablup/backend.ai#54 for more details.

### CODE REVIEW

File: src/ai/backend/gateway/stream.py [hunk 1]
```diff
@@ -573,6 +599,7 @@
     app['stream_execute_handlers'] = defaultdict(weakref.WeakSet)
     app['stream_proxy_handlers'] = defaultdict(weakref.WeakSet)
     app['stream_stdin_socks'] = defaultdict(weakref.WeakSet)
+    app['stream_extra_info'] = defaultdict(weakref.WeakSet)
```
Comment (achimnol) [inline src/ai/backend/gateway/stream.py]: This seems not to be used elsewhere. Please remove it if so.

File: src/ai/backend/manager/registry.py [hunk 2]
```diff
@@ -843,7 +843,17 @@ def __init__(self, config_server, dbpool,
             async with RPCContext(kernel['agent_addr'], None) as rpc:
                 coro = rpc.call.start_service(str(kernel['id']), service, opts)
                 if coro is None:
-                    log.warning('stat_service cancelled')
+                    log.warning('start_service cancelled')
+                    return None
+                return await coro
+
+    async def get_service_extra(self, sess_id, access_key, service):
```
Comment (achimnol) [inline src/ai/backend/manager/registry.py]: Rename to `get_service_ports()`.

File: src/ai/backend/gateway/stream.py [hunk 3]
```diff
@@ -388,6 +390,11 @@
 
     try:
         opts: Mapping[str, Any] = {}
+        if params['arguments'] != None:
```
Comment (achimnol) [inline src/ai/backend/gateway/stream.py]: Use `is not` instead of `!=`.
See https://stackoverflow.com/questions/3257919/what-is-the-difference-between-is-none-and-none
TL;DR: PEP-8 recommends `is` operator for `None` comparison and it is much faster.

File: src/ai/backend/gateway/stream.py [hunk 4]
```diff
@@ -322,6 +322,8 @@
     t.Dict({
         tx.AliasedKey(['app', 'service']): t.String,
         tx.AliasedKey(['port'], default=None): t.Null | t.Int[1024:65535],
+        tx.AliasedKey(['arguments'], default=None): t.Null | t.String, # {'-P': '12345'}
+        tx.AliasedKey(['envs'], default=None): t.Null | t.String # {'PASSWORD': '12345'}
```
Comment (achimnol) [inline src/ai/backend/gateway/stream.py]: Insert an additional space before the comment sign (`#`) as PEP-8 recommends to use *two or more* spaces before line-ending comments.

File: src/ai/backend/gateway/stream.py [hunk 5]
```diff
@@ -322,6 +322,8 @@
     t.Dict({
         tx.AliasedKey(['app', 'service']): t.String,
         tx.AliasedKey(['port'], default=None): t.Null | t.Int[1024:65535],
+        tx.AliasedKey(['arguments'], default=None): t.Null | t.String, # {'-P': '12345'}
```
Comment (achimnol) [inline src/ai/backend/gateway/stream.py]: Please include more formal definition of the value format like:
```suggestion
        tx.AliasedKey(['arguments'], default=None): t.Null | t.String,  # stringified JSON, e.g., '{"-P": "12345"}'
                                                                        # For flag-style arguments, the value must be None.
```

File: src/ai/backend/gateway/stream.py (hunk 4 above)
Comment (achimnol) [inline src/ai/backend/gateway/stream.py]: Please include more formal definition of the value format like:
```suggestion
        tx.AliasedKey(['envs'], default=None): t.Null | t.String     # stringified JSON, e.g., '{"PASSWORD": "12345"}'
```

**Comment by kyujin-cho** [comment]
Published new commit which contains:
- agent will now return information of services which are defined in local docker images, but not in etcd
  - Priority of each service definition sources will be: intrinsic > etcd > local docker image label
- agent's `service_info` field(inside `create_kernel` call's response data) now contains service port's template info by agent querying live service info right after kernel creation
- stream's `/apps` API will now fetch service info from psql; This means no RPC calls are made when querying service info
- `/apps/{app_name}` API is removed
- `get_service_apps` RPC call is removed, since we don't use it anymore
